File: src/main/java/com/chattriggers/ctjs/triggers/OnSoundPlayTrigger.java
package com.chattriggers.ctjs.triggers;

import lombok.Setter;

public class OnSoundPlayTrigger extends OnTrigger {
    @Setter
    private String soundNameCriteria = "";

    public OnSoundPlayTrigger(Object method) {
        super(method, TriggerType.SOUND_PLAY);
    }

    /**
     * Sets the sound name criteria.<br>
     * Short hand for {@link OnSoundPlayTrigger#setSoundNameCriteria(String)}.
     *
     * @param soundNameCriteria the sound name
     * @return the trigger for method chaining
     */
    public OnSoundPlayTrigger setCriteria(String soundNameCriteria) {
        this.soundNameCriteria = soundNameCriteria;

        return this;
    }

    @Override
    public void trigger(Object... args) {
        if (args[2] instanceof String
        && !soundNameCriteria.equals("")
        && !((String) args[2]).equalsIgnoreCase(soundNameCriteria))
            return;

        callMethod(args);
    }
}

File: src/main/kotlin/com/chattriggers/ctjs/triggers/OnSoundPlayTrigger.kt
package com.chattriggers.ctjs.triggers

import com.chattriggers.ctjs.engine.ILoader

class OnSoundPlayTrigger(method: Any, loader: ILoader) : OnTrigger(method, TriggerType.SoundPlay, loader) {
    private var soundNameCriteria = ""

    /**
     * Sets the sound name criteria.
     *
     * @param soundNameCriteria the sound name
     * @return the trigger for method chaining
     */
    fun setCriteria(soundNameCriteria: String) = apply { this.soundNameCriteria = soundNameCriteria }

    override fun trigger(args: Array<out Any?>) {
        if (args[1] is String
            && soundNameCriteria != ""
            && !(args[1] as String).equals(soundNameCriteria, ignoreCase = true)
        )
            return

        callMethod(args)
    }
}
